src/manufactured_solns.jl

```diff
@@ -225,5 +230,38 @@ using Symbolics
         
         return manufactured_sources_list
     end 
-    
+
+    """
+        manufactured_solutions_as_arrays(
+            t::mk_float, r::AbstractVector, z::AbstractVector, vperp::AbstractVector,
+            vpa::AbstractVector)
+
+    Create array filled with manufactured solutions.
+
+    Returns
+    -------
+    (densi, phi, dfni)
+    """
+    function manufactured_solutions_as_arrays(

```

Comment:
Does a function like this make it possible to avoid passing the MMS functions around the main code?

If I not mistaken, there appears to be a speed cost associated with passing the runtimegenerated function -- it would be interesting to know if you notice a significant speedup by using this approach instead.

Reply:
I don't think there would be, mostly because this function allocates new arrays for its output, then returns them. Also it's re-building the manufactured-solutions functions each time it's called.

It should be possible to set up a struct in a way that everything in the main time-loop is fully typed. I'll have a look now and check if there is some type instability...